matchms/filtering/add_precursor_mz.py

```diff
@@ -12,10 +25,16 @@ def add_precursor_mz(spectrum_in: SpectrumType) -> SpectrumType:
 
     spectrum = spectrum_in.clone()
 
-    if isinstance(spectrum.get("precursor_mz", None), str):
-        spectrum.set("precursor_mz", float(spectrum.get("precursor_mz").strip()))
-    elif spectrum.get("precursor_mz", None) is None:
-        pepmass = spectrum.get("pepmass", None)
+    precursor_mz_key = get_first_common_element(spectrum.metadata.keys(), _accepted_keys)
+    precursor_mz = spectrum.get(precursor_mz_key)
+
+    if isinstance(precursor_mz, str):
+        precursor_mz = float(precursor_mz.strip())
```

Comment:
We will always keep some non-covered corner cases I guess. Currently it will fail for string entries such as `"n/a"`. Not sure we have to cover that.
(feel free to skip this one until someone really has this issue)

Reply:
I implemented some basic checking if the string can be converted, this should cover most non-sensical values and it also prints an error message.